src/muse/utilities.py

```diff
@@ -141,30 +141,40 @@ def reduce_assets(
     """
     from copy import copy
 
+    assets = copy(assets)
```

Comment:
Is it copy enough and not necessary deepcopy?

Reply:
It doesn't seem to make a difference, but I'll switch to deepcopy to be on the safe side